src/ReinforcementLearningEnvironments/src/environments/examples/examples.jl

```diff
@@ -11,3 +11,7 @@ include("CartPoleEnv.jl")
 include("MountainCarEnv.jl")
 include("PendulumEnv.jl")
 include("BitFlippingEnv.jl")
+
+# checking the state of env is enough?
+Base.:(==)(env1::AbstractEnv, env2::AbstractEnv) = state(env1) == state(env2)
+Base.hash(env::AbstractEnv, h::UInt) = hash(state(env), h)
```

Comment:
Maybe we can move this into RLBase?

By the way, you can restrict the type to be of the same env type like this:

```suggestion

# checking the state of env is enough?
Base.:(==)(env1::T, env2::T) where T<:AbstractEnv = state(env1) == state(env2)
Base.hash(env::AbstractEnv, h::UInt) = hash(state(env), h)
```

Also add a doc string to warn users that the we only check the state instead of the game specification part.

Reply:
Thank you for your kind advice. I'll push changes later.